app/scripts/components/globalComponents/metadataObjectCreator.jsx

```diff
@@ -0,0 +1,347 @@
+                        Properties
+                    </h5> :
+                    <Paper className="mdl-cell mdl-cell--12-col"
+                           style={styles.warningDiv}
+                           zDepth={1}>
+                        <span>You must add properties to this template before you can use it.
+                        Go to the Annotations option in the main side menu at the top left of the screen to do this.
+                    </span>
+                    </Paper>
+                        }
+                </div>
+                <div className="mdl-cell mdl-cell--12-col mdl-color-text--grey-600" style={styles.listWrapper}>
+                    {this.props.templateProperties && this.props.templateProperties.length ? <Table selectable={false}>
+                        <TableHeader displaySelectAll={false} adjustForCheckbox={false}>
+                            <TableRow>
+                                <TableHeaderColumn style={styles.tableHead1}>Key</TableHeaderColumn>
+                                <TableHeaderColumn style={styles.tableHead2}>Datatype</TableHeaderColumn>
+                                <TableHeaderColumn style={styles.tableHead3}>Value</TableHeaderColumn>
+                            </TableRow>
+                        </TableHeader>
+                        <TableBody displayRowCheckbox={false}>
+                                {properties}
+                        </TableBody>
+                    </Table> : null}
+                </div>
+                <div className="mdl-cell mdl-cell--12-col" style={styles.btnWrapper}>
+                    <Paper className="mdl-cell mdl-cell--12-col"
+                           style={{backgroundColor: '#ef5350', color: '#EEEEEE', height: 40, marginBottom: 10, marginTop: 10, padding: 10, textAlign: 'center', display: showWarning}}
+                           zDepth={1}>
+                       <span>You must add at least one value</span>
+                    </Paper>
+                    <RaisedButton label={'Cancel'} secondary={true}
+                                  labelStyle={styles.button.label}
+                                  style={styles.button.cancel}
+                                  onTouchTap={() => this.toggleTagManager()}/>
+                    {this.props.templateProperties && this.props.templateProperties.length ?
+                    <RaisedButton label={'Apply'} secondary={true}
+                                  labelStyle={styles.button.label}
+                                  style={styles.button.apply}
+                                  onTouchTap={() => this.createMetadataObject(templateId)}/> : null}
+                </div>
+            </div>
+        )
+    }
+
+    addDateProperty(x, date, id) { // x is event which is always null. This is MUI behavior
+        if(!BaseUtils.objectPropInArray(this.state.metaProps, 'key', id)) { //If not in array, add object
+            this.state.metaProps.push({key: id, value: date});
```

Comment:
Is there a reason why you are altering state directly and not using `this.setState({metaProps})`? And also, is there a reason why you're using local state vs storing this in a store?